shim/linux/src/entry.c

```diff
@@ -788,10 +789,24 @@ dispatch_vcpu_kvm_get_one_reg(struct kvm_one_reg *const ioctl_args)
 }
 
 static long
-dispatch_vcpu_kvm_get_regs(struct kvm_regs *const ioctl_args)
+dispatch_vcpu_kvm_get_regs(
+    struct shim_vcpu_t const *const vcpu, struct kvm_regs *const regs)
 {
-    (void)ioctl_args;
-    return -EINVAL;
+    int64_t ret;
+    struct kvm_regs i_kvm_regs;
```

Comment:
since KVM doesn't use _t like the rest of the world for struct types, variable names becomes an issue, which is likely why you added "i_" to the variable name. We should adopt a common scheme for this. Every kvm_xxx struct will be the only args passed back and forth, so I think what we should do is the following:

```
dispatch(kvm_xxx *user_args)
{
    kvm_xxx args;
    copy_from(&args, user_args, ...)
    handle_xxx(&args);
    copy_to(user_args, &args, ...)
}
```

This way, each dispatch function has a similar feel, and it describes better what is actually going on here. So the TL;DR is change `regs` to `user_args` and `i_kvm_regs` to `args`.